### ISSUE TITLE
Bug in `wts` with `avg_predictions()` and `by`

### ISSUE THREAD

**Issue by ngreifer** [vincentarelbundock/marginaleffects#695]
Trying to submit a new version of `MatchIt` to CRAN that uses the new `avg_()` functions in the vignettes. Everything works fine, except for when I try to get adjusted predictions with weighted subsetted data. I only get this error on CRAN, not on my own computer, which suggests an environment problem. Below is the error I get from CRAN:

> --- re-building 'estimating-effects.Rmd' using rmarkdown
> Quitting from lines 246-251 (estimating-effects.Rmd) 
> Error: processing vignette 'estimating-effects.Rmd' failed with diagnostics:
> The `wts` argument must be a numeric vector of length 882, or a string which matches a column name in `newdata`. If you did not supply a `newdata` explicitly, `marginaleffects` extracted it automatically from the model object, and the `wts` variable may not have been available. The easiest strategy is often to supply a data frame such as the original data to `newdata` explicitly, and to make sure that it includes an appropriate column of weights, identified by the `wts` argument.
> 

These lines refer to the following call:
```r
avg_predictions(fit1, variables = "A",
                vcov = ~subclass,
                newdata = subset(md, A == 1),
                wts = "weights",
                by = "A")
```

This does not occur with the `avg_comparisons()` call right above it, which is

```r
avg_comparisons(fit1, variables = "A",
                vcov = ~subclass,
                newdata = subset(md, A == 1),
                wts = "weights")
```
and which runs fine.

I can reproduce the error only using win_builder. It also occurs when I omit the `newdata` line. This is using version 0.9.0.

**Comment by vincentarelbundock** [comment]
Thanks for the report. I tracked this down to a rather subtle bug related to the order in which I run sanity checks on the inputs. It should be fixed in c9a86ce. I don't remember touching this code recently, so I'm kind of surprised that this ever worked... ¯\\\_(ツ)_\/¯

In any case, I added your vignette example to the test suite, so it should be more future proof. Feel free to add your own test cases here as a safety net for critical examples: https://github.com/vincentarelbundock/marginaleffects/blob/main/inst/tinytest/test-pkg-MatchIt.R

The bad news is that I *just* sent version 0.10.0 to CRAN. It's the 2nd release in quick succession, because I had to fix a critical issue with standard errors. Unfortunately, this means that this particular `predictions()` fix probably won't reach CRAN until next month.

Perhaps you could make evaluation of that code block conditional on `packageVersion("marginaleffects")>"0.10.0")` and include a short note for readers?

Very sorry for the inconvenience!

**Comment by ngreifer** [comment]
Thanks for looking into it! I think I'll make evaluation conditional since MatchIt is in peril right now and I need to submit it.
